frontend/src/views/Collection/index.tsx

```diff
@@ -102,6 +103,13 @@ const Collection: FC<Props> = ({ id = "" }) => {
   const datasetPresent =
     collection.datasets?.length > 0 || Object.keys(uploadedFiles).length > 0;
 
+  const invalidateCollectionQuery = memoize(
```

Comment:
We need to memoize `invalidateCollectionQuery` here, since `<DatasetsGrid />` is shallow diffing (`===`) it to decide whether it needs to re-render

If we don't memoize the function, the following code will always generate a new function and therefore `props.invalidateCollectionQuery === prevProps.invalidateCollectionQuery` will always be `false`

```tsx
invalidateCollectionQuery={() => {
              queryCache.invalidateQueries([USE_COLLECTION, id, visibility]);
            }}
```